autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingHandler.java

```diff
@@ -48,30 +50,37 @@
 
     @Override
     public void onApplicationEvent(ScalingEvent event) {
-        BaseAlert alert = event.getAlert();
-        Cluster cluster = clusterService.findById(alert.getCluster().getId());
+        Set<BaseAlert> alerts = event.getAlerts();
+        Long clusterId = alerts.stream().findFirst().map(ma -> ma.getCluster().getId()).orElseThrow();
+        Cluster cluster = clusterService.findById(clusterId);
         MDCBuilder.buildMdcContext(cluster);
-        scale(cluster, alert.getScalingPolicy());
+        if (isCooldownElapsed(cluster)) {
+            alerts.forEach(alert -> scale(cluster, alert.getScalingPolicy()));
+        }
     }
 
     private void scale(Cluster cluster, ScalingPolicy policy) {
+        int totalNodes = Math.toIntExact(cloudbreakClient.autoscaleEndpoint().getHostMetadataCountForAutoscale(cluster.getStackId(), policy.getHostGroup()));
+        int desiredNodeCount = getDesiredNodeCount(cluster, policy, totalNodes);
+        if (totalNodes != desiredNodeCount) {
+            Runnable scalingRequest = (Runnable) applicationContext.getBean("ScalingRequest", cluster, policy, totalNodes, desiredNodeCount);
+            loggedExecutorService.submit("ScalingHandler", scalingRequest);
+            rejectedThreadService.remove(cluster.getId());
+            cluster.setLastScalingActivityCurrent();
+            clusterService.save(cluster);
+        } else {
+            LOGGER.info("No scaling activity required");
```

Comment:
The log message could be enhanced and contain the name of the alert/scaling action pair for the sake of easier debugging in the future.

Reply:
this information is available in MDC context. But it is not enough, I can update the message